### DISCUSSION TITLE
Fetching 3rd party login API after obtaining token from OAuth provider

### DISCUSSION THREAD

**Discussion by mwskwong** [nextauthjs/next-auth#6807] [Help]
I'm hoping to achieve the following workflow:
1. Login through an OAuth provider (Let's say it's GitHub, just to make things easy) and obtain the token
2. With the token in hand, use it as part of the request body to request login with a 3rd party backend. 
    The backend is black-boxed, and I assume it will return a new token as well as the user object.
3. The entire Next.js application is based around this new token and session.

My question is, how should I implement step (2)? 

I can get OAuth working and have access to the session object of it in the client. But I'm hoping to get the session object of the 3rd party backend.

**Comment by AndonMitev** [comment]
hey did you find any solution? not sure how to read the access token in the service for data fetching
